src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -2032,15 +2046,26 @@ FixedwingPositionControl::control_position(const math::Vector<2> &current_positi
 
 	}
 
-	/* During a takeoff waypoint while waiting for launch the pitch sp is set
-	 * already (not by tecs) */
-	if (!(_control_mode_current ==  FW_POSCTRL_MODE_AUTO && (
-		      (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF &&
-		       (launch_detection_state == LAUNCHDETECTION_RES_NONE ||
-			_runway_takeoff.runwayTakeoffEnabled())) ||
-		      (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LAND &&
-		       land_noreturn_vertical))
-	     )) {
+	// decide when to use pitch setpoint from TECS because in some cases pitch
+	// setpoint is generated by other means
+	bool use_tecs_pitch = true;
+
+	// auto takeoff
+	use_tecs_pitch &= !(_control_mode_current ==  FW_POSCTRL_MODE_AUTO &&
+		      pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF &&
+		       launch_detection_state == LAUNCHDETECTION_RES_NONE);
+
+	// runway takeoff
+	use_tecs_pitch &= !_runway_takeoff.runwayTakeoffEnabled();
```

Comment:
_runway_takeoff.runwayTakeoffEnabled() is checking if the param is set.
So this would also need to be only during AUTO with a TAKEOFF setpoint.

Overall you've made that nasty check almost readable ;)